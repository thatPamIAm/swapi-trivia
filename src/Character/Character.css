/*Base styling for Character in Movie.css*/
/*Additional styling below*/

.character-films-container {
  height: 155px;
}

li {
  list-style-type: none;
}

span {
  font-weight: bold;
}

.favorites-btn {
  height: 40px;
  width: 150px;
}