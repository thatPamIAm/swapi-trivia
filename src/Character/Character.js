import React, { Component } from 'react';

import './Character.css'

class Character extends Component {
  constructor(props){
    super(props);
    this.state = {
      homeworld: '',
      population: 0,
      species: '',
      films: []
    }
  }

  componentDidMount() {
    this.fetchHomeworld(this.props.character.homeworld);
    this.fetchSpecies(this.props.character.species[0]);
    this.fetchFilms(this.props.character.films);
  }

  async fetchHomeworld(url) {
    let data = await fetch(url);
    data = await data.json();

    this.setState({
      homeworld: data.name,
      population: data.population
    });
  }

  async fetchSpecies(url) {
    let data = await fetch(url);
    data = await data.json();

    this.setState({
      species: data.name
    })
  }

  async fetchFilms(list) {
    let films = await Promise.all(
      list.map(async url => {
        let film = await fetch(url);
        return film.json();
      })
    )
    console.log(films)

    this.setState({
      films
    })
  }

  renderFilms() {
    return this.state.films.map((film, index) => {
      return <li key={film.release_date}>{film.title}</li>
    });
  }

  render() {
    const { name } = this.props.character;
    const { homeworld, population, species } = this.state;
 

    return(
      <article className="card">
        <h2>{name}</h2>
        <li><span>Homeworld: </span>{homeworld}</li>
        <li><span>Population: </span>{population}</li>
        <li><span>Species: </span>{species}</li>
        <li><span>Films:</span></li>
        <div className="character-films-container">
          {this.renderFilms()}
        </div>
        <button className="favorites-btn">Favorite</button>
      </article>
    ) 
  }
}

export default Character;