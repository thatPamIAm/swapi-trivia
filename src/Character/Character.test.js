import React from 'react';
import { shallow } from 'enzyme';

import Character from './Character';

describe('Character', () => {
  const mockCharacter = {
    name: 'C-3PO',
    homeworld: 'https://swapi.co/api/planets/1/',
    species:'https://swapi.co/api/species/2/',
    films: ['https://swapi.co/api/films/2/']
  }
  
  const mockIndex = 0;

  it('should match the snapshot', () => {
    let wrapper = shallow(<Character character={mockCharacter} key={mockIndex} />)
    expect(wrapper).toMatchSnapshot();
  });
});