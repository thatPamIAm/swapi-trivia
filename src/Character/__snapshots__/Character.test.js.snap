// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match the snapshot 1`] = `ShallowWrapper {}`;
